packages/docs/api/equality-algorithm.md

```diff
@@ -0,0 +1,225 @@
+---
+
+The equality algorithm is an extensible and parameterized deep equality checker.
+It is used by all functions that check deep equality inside earl.
+
+## Algorithm
+
+To determine is two values are equal the following checks are executed:
+
+1. If the second value is a [Matcher](/guides/using-matchers.md) and it...
+   1. matches the first value, then the values are considered equal;
+   1. doesn't match the first value, then the values are considered unequal.
+1. Every [custom rules](/advanced/plugin-development.md) is checked.
+   1. If it returns undefined the next rule is checked or the algorithm
+      advances.
+   1. If it returns success, then the values are considered equal.
+   1. If it returns error, then the values are considered unequal.
+1. If any of the values is equal to one of it's ancestors (is a self-referencing
+   object)...
+   1. If the other value is equal to it's ancestor of the same distance (e.g.
+      grandparent and grandparent), then the values are considered equal.
+   2. Otherwise they are considered unequal.
+1. A category is determined for each value based on the rules of each category.
+   1. If the values have different categories, then they are considered unequal.
+   1. If the category is the same the values are compared according to rules of
+      that category.
+
+Additionally the algorithm takes the following options:
+
+- `uniqueNaNs` - Considers two `NaN` values unequal, defaults to `false`.
+- `minusZero` - Considers `+0` and `-0` unequal, defaults to `false`.
+- `ignorePrototypes` - Considers `new Vector2(1, 2)` and `{ x: 1, y: 2 }` equal,
+  defaults to `false`. This option is set to `true` when using
+  [`.toLooseEqual`](/api/api-reference.md#toLooseEqual-value-any).
+- `compareErrorStack` - Compares the stack property of `Error` instances,
+  defaults to `false`.
+
+## Categories
+
+### Primitive
+
+A value belongs to the Primitive category if `typeof value` returns any of the
+following:
+
+- `"undefined"`
+- `"boolean"`
+- `"number"`
+- `"bigint"`
+- `"string"`
+- `"symbol"`
+- `"object"`, but only if the value is `null`
+
+The equality of values belonging to the Primitive category is determined in the
+following way:
+
+1. If both values are `NaN` and `uniqueNaNs` is set to `false`, then the values
+   are considered equal.
+1. If `minusZero` is set to `true` then `0` is only considered equal to `0` and
+   `-0` is only considered equal to `-0`.
+1. Otherwise it is determined by the `===` operator.
+
+### Function, Promise, WeakMap, WeakSet
+
+A value belongs to the Function category if `typeof value` returns `"function"`.
+
+A value belongs to the Promise category if `value instanceof Promise` returns
+`true`.
+
+A value belongs to the WeakMap category if `value instanceof WeakMap` returns
+`true`.
+
+A value belongs to the WeakSet category if `value instanceof WeakSet` returns
+`true`.
+
+The equality of values belonging to those categories is determined by the `===`
+operator.
```

Comment:
```suggestion
- A value belongs to the Function category if `typeof value` returns `"function"`.
- A value belongs to the Promise category if `value instanceof Promise` returns
`true`.
- A value belongs to the WeakMap category if `value instanceof WeakMap` returns
`true`.
- A value belongs to the WeakSet category if `value instanceof WeakSet` returns
`true`.

The equality of values belonging to those categories is determined by the `===`
operator.
```

IMHO these would be more readable as bullet points.

Reply:
:+1: